### PR TITLE
feat: add llm.requestDelay setting v2

### PR DESCRIPTION
**Pull request by antoinejeannot**
Hi @McPatate 👋 

This PR is based on https://github.com/huggingface/llm-vscode/pull/103 by @davidpissarra which looks stale without further improvements / changes for a month, but which brings a really neat feature imo :)

As the original PR, it adds the `llm.requestDelay` setting, representing the minimum time interval between requests, in __milliseconds__ though. It now defaults to `150ms` in addition to taking into account the cancellation token.

Hope this helps! I would not mind declining this PR if any changes happen to be made on the other one.
Thanks for your review

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -297,4 +307,34 @@ export default async function highlightStackAttributions(): Promise<void> {
...
+	 * @param token cancellation token
+	 * @returns a promise that resolves after N milliseconds, unless the token is cancelled.
+	 *
+	 * @remarks This is a workaround for the lack of a debounce function in vscode.
+	*/
+    return new Promise((resolve, reject) => {
+        const interval = setInterval(() => {
+            if (token.isCancellationRequested) {
+                clearInterval(interval);
+                reject(new Error('Operation cancelled'));
```
Comment (McPatate) [inline src/extension.ts:327]: Just a nit, I'd rather return a `cancelled: bool` than an error.

**Review by McPatate — COMMENTED** [review]
Thank you for your contribution! Lgtm, though small nit on the API.

File: src/extension.ts (hunk 1 above)
Comment (antoinejeannot) [reply src/extension.ts]: I refactored it so that the promise now resolves to a boolean.
Is that what you have in mind ?

**Comment by antoinejeannot** [comment]
Thanks for your review @McPatate, PTAL

**Comment by antoinejeannot** [comment]
I have batched all conversations to write the latest commit. Thanks for your input @McPatate

**Review by McPatate — APPROVED** [review]
lgtm, thank you for your work!

**Pull request merged**